plugins/Time/local/ddate.py

```diff
@@ -0,0 +1,175 @@
+# copied from https://github.com/a-tal/ddate
```

Comment:
Is there a reason why you embed ddate instead of using it as an optional dependency? It is available on [PyPI](https://pypi.python.org/pypi/ddate/) after all.

Reply:
to allow use from install, without anything else being installed

Reply:
and this notice does not follow ddate's license:

```
Redistributions of source code must retain the above copyright notice, this
list of conditions and the following disclaimer.
```

Reply:
then I'll copy that over

Reply:
I would still prefer an optional dependency

Reply:
I would still prefer an optional dependency, unless you see an issue with that.

Reply:
Seriously though, I don't agree with embedding dependencies simply for the sake of doing so. In the long run it's more maintenance effort and makes makes tracking fixes, etc. harder. Distros are very much against this practice too for good reason:

- https://wiki.gentoo.org/wiki/Why_not_bundle_dependencies
- https://wiki.debian.org/EmbeddedCodeCopies

Reply:
I'll change it to an optional dependency

Reply:
@GLolol None of these arguments apply to such a simple library, though.